File: Creational/Builder/C++/lib/ConcreteBuilder2.cpp
#include "../inc/ConcreteBuilder2.hpp"
#include "../inc/ConcreteProduct2.hpp"

ConcreteBuilder2::ConcreteBuilder2()
{

}

ConcreteBuilder2::~ConcreteBuilder2()
{

}

void ConcreteBuilder2::reset()
{
	m_product = std::make_shared<ConcreteProduct2>();
}

void ConcreteBuilder2::build_step_a()
{
	m_product->step_a_2();
}

void ConcreteBuilder2::build_step_b()
{
	m_product->step_b_2();
}

void ConcreteBuilder2::build_step_c()
{
	m_product->step_c_2();
}

File: Creational/Builder/Python/ConcreteBuilder2.py
from AbstractBuilder import *
from ConcreteProduct2 import *

class ConcreteBuilder2(AbstractBuilder):

	def reset(self):
		self.product = ConcreteProduct2()
		return 
		
	def build_step_a(self):
		self.product.step_a_2()
		
	def build_step_b(self):
		self.product.step_b_2()
		
	def build_step_c(self):
		self.product.step_c_2()